source/common/upstream/upstream_impl.cc

```diff
@@ -1724,32 +1736,41 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
   const bool dont_remove_healthy_hosts =
       health_checker_ != nullptr && !info()->drainConnectionsOnHostRemoval();
   if (!current_priority_hosts.empty() && dont_remove_healthy_hosts) {
-    erase_from =
```

Comment:
Is it possible to avoid this reformatting? I think it would be good to avoid the git history churn here

Reply:
done

Reply:
oh, looks like ci wants formatting:

```
021-12-13T23:53:11.8379276Z diff --git a/source/common/upstream/upstream_impl.cc b/source/common/upstream/upstream_impl.cc
2021-12-13T23:53:11.8380398Z index 2e4e84b2d..87c6b3196 100644
2021-12-13T23:53:11.8384499Z --- a/source/common/upstream/upstream_impl.cc
2021-12-13T23:53:11.8385175Z +++ b/source/common/upstream/upstream_impl.cc
2021-12-13T23:53:11.8385947Z @@ -1736,42 +1736,41 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
2021-12-13T23:53:11.8386543Z    const bool dont_remove_healthy_hosts =
2021-12-13T23:53:11.8387232Z        health_checker_ != nullptr && !info()->drainConnectionsOnHostRemoval();
2021-12-13T23:53:11.8387824Z    if (!current_priority_hosts.empty() && dont_remove_healthy_hosts) {
2021-12-13T23:53:11.8388431Z -    erase_from =
2021-12-13T23:53:11.8389152Z -        std::remove_if(current_priority_hosts.begin(), current_priority_hosts.end(),
2021-12-13T23:53:11.8389939Z -                       [&all_new_hosts, &new_hosts_for_current_priority,
2021-12-13T23:53:11.8390787Z -                        &hosts_with_updated_locality_for_current_priority, &final_hosts,
2021-12-13T23:53:11.8391583Z -                        &max_host_weight](const HostSharedPtr& p) {
2021-12-13T23:53:11.8392501Z -                         // This host has already been added as a new host in the
2021-12-13T23:53:11.8395844Z -                         // new_hosts_for_current_priority. Return false here to make sure that host
2021-12-13T23:53:11.8397375Z -                         // reference with older locality gets cleaned up from the priority.
2021-12-13T23:53:11.8398850Z -                         if (hosts_with_updated_locality_for_current_priority.contains(
2021-12-13T23:53:11.8400187Z -                                 p->address()->asString())) {
2021-12-13T23:53:11.8401055Z -                           return false;
2021-12-13T23:53:11.8401878Z -                         }
2021-12-13T23:53:11.8402351Z -
2021-12-13T23:53:11.8403698Z -                         if (all_new_hosts.contains(p->address()->asString()) &&
2021-12-13T23:53:11.8404634Z -                             !new_hosts_for_current_priority.contains(p->address()->asString())) {
2021-12-13T23:53:11.8405475Z -                           // If the address is being completely deleted from this priority, but is
2021-12-13T23:53:11.8406284Z -                           // referenced from another priority, then we assume that the other
2021-12-13T23:53:11.8407246Z -                           // priority will perform an in-place update to re-use the existing Host.
2021-12-13T23:53:11.8408214Z -                           // We should therefore not mark it as PENDING_DYNAMIC_REMOVAL, but
2021-12-13T23:53:11.8409106Z -                           // instead remove it immediately from this priority.
2021-12-13T23:53:11.8410265Z -                           // Example: health check address changed and priority also changed
2021-12-13T23:53:11.8411154Z -                           return false;
2021-12-13T23:53:11.8411957Z -                         }
2021-12-13T23:53:11.8412487Z -
2021-12-13T23:53:11.8413803Z -                         if (!(p->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC) ||
2021-12-13T23:53:11.8415695Z -                               p->healthFlagGet(Host::HealthFlag::FAILED_EDS_HEALTH))) {
2021-12-13T23:53:11.8416691Z -                           if (p->weight() > max_host_weight) {
2021-12-13T23:53:11.8417555Z -                             max_host_weight = p->weight();
2021-12-13T23:53:11.8419015Z -                           }
2021-12-13T23:53:11.8419880Z -
2021-12-13T23:53:11.8420517Z -                           final_hosts.push_back(p);
2021-12-13T23:53:11.8421304Z -                           p->healthFlagSet(Host::HealthFlag::PENDING_DYNAMIC_REMOVAL);
2021-12-13T23:53:11.8423123Z -                           return true;
2021-12-13T23:53:11.8423870Z -                         }
2021-12-13T23:53:11.8424480Z -                         return false;
2021-12-13T23:53:11.8425030Z -                       });
2021-12-13T23:53:11.8425807Z +    erase_from = std::remove_if(
2021-12-13T23:53:11.8426236Z +        current_priority_hosts.begin(), current_priority_hosts.end(),
2021-12-13T23:53:11.8426861Z +        [&all_new_hosts, &new_hosts_for_current_priority,
2021-12-13T23:53:11.8427394Z +         &hosts_with_updated_locality_for_current_priority, &final_hosts,
2021-12-13T23:53:11.8427795Z +         &max_host_weight](const HostSharedPtr& p) {
2021-12-13T23:53:11.8428329Z +          // This host has already been added as a new host in the
2021-12-13T23:53:11.8428798Z +          // new_hosts_for_current_priority. Return false here to make sure that host
2021-12-13T23:53:11.8429406Z +          // reference with older locality gets cleaned up from the priority.
2021-12-13T23:53:11.8430161Z +          if (hosts_with_updated_locality_for_current_priority.contains(p->address()->asString())) {
2021-12-13T23:53:11.8430673Z +            return false;
2021-12-13T23:53:11.8431017Z +          }
2021-12-13T23:53:11.8431393Z +
2021-12-13T23:53:11.8431954Z +          if (all_new_hosts.contains(p->address()->asString()) &&
2021-12-13T23:53:11.8432679Z +              !new_hosts_for_current_priority.contains(p->address()->asString())) {
2021-12-13T23:53:11.8433262Z +            // If the address is being completely deleted from this priority, but is
2021-12-13T23:53:11.8433831Z +            // referenced from another priority, then we assume that the other
2021-12-13T23:53:11.8434762Z +            // priority will perform an in-place update to re-use the existing Host.
2021-12-13T23:53:11.8435466Z +            // We should therefore not mark it as PENDING_DYNAMIC_REMOVAL, but
2021-12-13T23:53:11.8436022Z +            // instead remove it immediately from this priority.
2021-12-13T23:53:11.8436555Z +            // Example: health check address changed and priority also changed
2021-12-13T23:53:11.8436961Z +            return false;
2021-12-13T23:53:11.8437308Z +          }
2021-12-13T23:53:11.8437617Z +
2021-12-13T23:53:11.8438639Z +          if (!(p->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC) ||
2021-12-13T23:53:11.8439731Z +                p->healthFlagGet(Host::HealthFlag::FAILED_EDS_HEALTH))) {
2021-12-13T23:53:11.8440438Z +            if (p->weight() > max_host_weight) {
2021-12-13T23:53:11.8441245Z +              max_host_weight = p->weight();
2021-12-13T23:53:11.8441815Z +            }
2021-12-13T23:53:11.8442110Z +
2021-12-13T23:53:11.8442389Z +            final_hosts.push_back(p);
2021-12-13T23:53:11.8442994Z +            p->healthFlagSet(Host::HealthFlag::PENDING_DYNAMIC_REMOVAL);
2021-12-13T23:53:11.8443665Z +            return true;
2021-12-13T23:53:11.8443961Z +          }
2021-12-13T23:53:11.8444579Z +          return false;
2021-12-13T23:53:11.8444904Z +        });
2021-12-13T23:53:11.8445305Z      current_priority_hosts.erase(erase_from, current_priority_hosts.end());
2021-12-13T23:53:11.8445646Z    }
2021-12-13T23:53:11.8445937Z  
2021-12-13T23:53:11.9481096Z ##[error]Bash exited with code '1'.
2021-12-13T23:53:11.9527811Z ##[section]Finishing: Run check format scripts
```

Reply:
do you think we can avoid the formatting?

Reply:
I guess this is fine then, just a bit confused why it wants to move std::remove_if to the next line.